src/Lean/Elab/PreDefinition/WF/GuessLex.lean

```diff
@@ -538,25 +553,40 @@ def mkTupleSyntax : Array Term → MetaM Term
 Given an array of `MutualMeasures`, creates a `TerminationWF` that specifies the lexicographic
 combination of these measures.
 -/
-def buildTermWF (declNames : Array Name) (varNamess : Array (Array Name))
-    (measures : Array MutualMeasure) : MetaM TerminationWF := do
+def buildTermWF (extraParams : Array Nat) (originalVarNamess : Array (Array Name))
+    (varNamess : Array (Array Name)) (measures : Array MutualMeasure) (trimVars : Bool)
+    : MetaM TerminationWF := do
   let mut termByElements := #[]
   for h : funIdx in [:varNamess.size] do
     let vars := (varNamess[funIdx]'h.2).map mkIdent
     let body ← mkTupleSyntax (← measures.mapM fun
       | .args varIdxs => do
-          let v := vars.get! (varIdxs[funIdx]!)
-          let sizeOfIdent := mkIdent (← unresolveNameGlobal ``sizeOf)
+          let varIdx := varIdxs[funIdx]!
```

Comment:
Could you please document the relationship between the different parameters in the `buildTermWF` doc-string?

Reply:
You rightfully called that function out, didn’t like it either. It’s now two functions, and better commented.